### PR TITLE
Terrain: Add light map support

### PR DESCRIPTION
**Pull request by super626**
And add interface Terrain::setLightDirection

### CODE REVIEW

File: cocos/3d/CCTerrain.h [hunk 1]
```diff
@@ -476,11 +479,13 @@ class CC_DLL Terrain : public Node
...
     Texture2D * _detailMapTextures[4];
     Texture2D * _alphaMap;
+    Texture2D * _lightMap;
+    Vec3 _lightDir;
     CustomCommand _customCommand;
     QuadTree * _quadRoot;
     Chunk * _chunkesArray[MAX_CHUNKES][MAX_CHUNKES];
     std::vector<TerrainVertexData> _vertices;
-    std::vector<GLushort> _indices;
+    std::vector<unsigned int> _indices;
```
Comment (dabingnn) [inline cocos/3d/CCTerrain.h:493]: It seems that `unsigned int index` could not work correctly on some old android device?
Could we keep using GLushort?

File: cocos/3d/CCTerrain.h [hunk 2]
```diff
@@ -304,6 +304,9 @@ class CC_DLL Terrain : public Node
     friend QuadTree;
     friend Chunk;
 public:
+    void setLightMap(const std::string& fileName);
+
+    void setLightDir(const Vec3& lightDir);
```
Comment (dabingnn) [inline cocos/3d/CCTerrain.h:314]: What space should be used when specify Light direction? World space, view space, or any other one?

File: cocos/3d/CCTerrain.h (hunk 1 above)
Comment (super626) [inline cocos/3d/CCTerrain.h:493]: This is used to compute normal, and short can be overflow when the number of trunk is huge. `_indices` is not going to be used during render.

File: cocos/3d/CCTerrain.h (hunk 2 above)
Comment (super626) [inline cocos/3d/CCTerrain.h:314]: It is in the terrain's local space. Terrain usually placed at (0, 0, 0) and without rotation, so most of the time we can say it is in world space.

File: cocos/3d/CCTerrain.h (hunk 1 above)
Comment (dabingnn) [inline cocos/3d/CCTerrain.h:493]: Thanks, please ignore it.

File: cocos/3d/CCTerrain.h (hunk 2 above)
Comment (dabingnn) [inline cocos/3d/CCTerrain.h:314]: Could you please add some comments to make it clearer?

**Pull request merged**